ocs_ci/ocs/machine.py

```diff
@@ -124,6 +124,20 @@ def get_machine_type(machine_name):
         break
 
 
+def get_machine_status(machine_obj):
+    """
+    Get the status of machine
+
+    Args:
+        machine_obj : machine obj
+
+    Returns:
+        str: The machine status (Running/Failed/Provisioned)
+
+    """
+    return machine_obj.get().get('status').get('phase')
```

Comment:
we have seen that in some cases, this will not return the up to date status. See https://bugzilla.redhat.com/show_bug.cgi?id=1715627
For that, as a workaround, we started getting the status from `oc get RESOURCE` without `-o yaml` and we parse the output. This is being done in `ocs_ci.ocs.ocp.OCP.get_resource_status`